File: CS 307 - Object Oriented Programming/RTTI/Simple RTTI Demonstration in C++/CShape.cpp
//======================================================================
// CShape.cpp
// Purpose: Implementation file defining a base class for this RTTI 
//				demonstration.
// Author: Dr. Rick Coleman
//======================================================================
#define _CRT_SECURE_NO_WARNINGS

#include "CShape.h"


//---------------------------------------
// Default constructor
//---------------------------------------
CShape::CShape()
{
	m_eType = SHAPE;
	strcpy(m_sTypeName, "CShape");
}

//---------------------------------------
// Default destructor
//---------------------------------------
CShape::~CShape()
{
}

//---------------------------------------
// Virtual function to return type
//---------------------------------------
SHAPE_TYPE CShape::getType()
{
	return m_eType;
}

//---------------------------------------
// Virtual function to return type name
//---------------------------------------
char *CShape::getTypeName()
{
	return m_sTypeName;
}

File: CS 307 - Object Oriented Programming/RTTI/RTTI in Java using getClass() and instanceof/CShape.java
//======================================================================
// CShape.java
// Purpose: Class file defining a base class for this RTTI demonstration.
// Author: Dr. Rick Coleman
//======================================================================

public class CShape
{
	/** Shape type */
	protected int m_iType;
	
	/** Class name */
	protected String m_sTypeName;
	
	public static final int SHAPE 		= 0;
	public static final int RECTANGLE 	= 1;
	public static final int OVAL 		= 2;
	public static final int LINE 		= 3;
	public static final int POLYGON 	= 4;
	
	//----------------------------------------------
	/** Default constructor */
	//----------------------------------------------
	public CShape()
	{
		m_iType = SHAPE;
		m_sTypeName = "CShape";
	}
	
	//----------------------------------------------
	/** Get the type */
	//----------------------------------------------
	public int getType()
	{
		return m_iType;
	}
	
	//----------------------------------------------
	/** Get the class name */
	//----------------------------------------------
	public String getTypeName()
	{
		return m_sTypeName;
	}
}